### PR TITLE
Test `document.currentScript` when script is moved during evaluation

### PR DESCRIPTION
**Pull request by hiroshige-g**
In the spec before https://github.com/whatwg/html/pull/5154,

`script element's node document` is re-evaluated after script evaluation,
and thus if a script is moved to another Document during its own evaluation,
`currentScript` of the *new* document could be updated.

After #5154, `currentScript` is always set/reset on the same Document,
which is expected, and tested so here.

Browser implementations already follow the new spec behavior (as expected).

### CODE REVIEW

File: html/semantics/scripting-1/the-script-element/moving-between-documents-during-evaluation.html [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      'Inner script should be evaluated synchronously');
+
+  assert_equals(document.currentScript, outerScript,
+      '[3] After inner script: currentScript of source Document');
+  assert_equals(window.anotherDocument.currentScript, null,
+      '[3] After inner script: currentScript of destination Document');
+
+  t.step_timeout(() => {
+    assert_equals(document.currentScript, null,
+        '[4] After outner script: currentScript of source Document');
```
Comment (domenic) [inline html/semantics/scripting-1/the-script-element/moving-between-documents-during-evaluation.html]: outner -> outer here and below

**Review by domenic — APPROVED** [review]
Wow, good spot. I will update the commit message to note that this is a fix, not editorial.

File: html/semantics/scripting-1/the-script-element/moving-between-documents-during-evaluation.html (hunk 1 above)
Comment (hiroshige-g) [reply html/semantics/scripting-1/the-script-element/moving-between-documents-during-evaluation.html]: Done.

**Comment by foolip** [comment]
@hiroshige-g there were Taskcluster troubles. If you try rebasing this PR it'll probably be fine now.

**Pull request merged**